custom_components/tahoma/coordinator.py

```diff
@@ -76,8 +76,13 @@ async def _async_update_data(self) -> Dict[str, Device]:
             raise UpdateFailed("Failed to connect.") from exception
         except (ServerDisconnectedError, NotAuthenticatedException):
             self.executions = {}
-            await self.client.login()
-            self.devices = await self._get_devices()
+
+            try:
+                await self.client.login()
+                self.devices = await self._get_devices()
+            except BadCredentialsException as exception:
```

Comment:
How it can be possible?

Reply:
I have seen this more often, however it happens in the following way:

1. Somfy has some kind of error and logs us out
2. We get a `NotAuthenticatedException` and will do a new login
3. New login will get a BadCredentialsException due to Somfy server issue (and retry a few times due to backoff)
4. We are blacklisted temporarily and throw a BadCredentialsException again here, that we don't catch
5. We keep retrying and the temporary ban will extend every time.